pygorithm/fibonacci/space_optimized.py

```diff
@@ -0,0 +1,52 @@
+# Author: Ashutosh Gupta
+# Created On: 8/15/2017
+# Time: 3:24 PM
+
+"""
+nth fibonacci number - Space Optimized
+"""
+
+import inspect
+
+
+def get_sequence(n):
+    """
+    Return Fibonacci sequence from zero to specified number as list.
+    """
+
+    def fib(n):
+        """
+        Return Fibonacci value by specified number as integer.
+        """
+        a = 0
+        b = 1
+
+        if n == 0:
+            return a
+
+        if n == 1:
+            return b
+
+        for i in range(1, n):
```

Comment:
You actually do not need `if` statements and `1` point at fist argument to `range`.

Make it simple like:
```python
    a, b= 0, 1

    for _ in range(n):
        a, b = b, a + b
    return a
```